edbterraform/data/terraform/aws/modules/machine/main.tf

```diff
@@ -136,7 +136,7 @@ resource "null_resource" "setup_volume" {
   provisioner "remote-exec" {
     inline = [
       "chmod a+x /tmp/setup_volume.sh",
-      "/tmp/setup_volume.sh ${element(local.string_device_names, tonumber(each.key))} ${each.value.mount_point} ${length(lookup(var.machine.spec, "additional_volumes", [])) + 1} ${coalesce(each.value.filesystem, local.filesystem)} >> /tmp/mount.log 2>&1"
+      "/tmp/setup_volume.sh ${element(local.string_device_names, tonumber(each.key))} ${each.value.mount_point} ${length(lookup(var.machine.spec, "additional_volumes", [])) + 1} ${coalesce(each.value.filesystem, local.filesystem)} ${coalesce(each.value.filesystemmountopt, local.filesystemmountopt)} >> /tmp/mount.log 2>&1"
```

Comment:
Join will fail with null so we will need to wrap it with a `try`:
```
> join(",", null)
╷
│ Error: Invalid function argument
│ 
│   on <console-input> line 1:
│   (source code not available)
│ 
│ Invalid value for "lists" parameter: argument must not be null.
╵
```
```
> try(join(",", null), null)
null
```

Reply:
ok